src/mds/Server.cc

```diff
@@ -585,7 +585,18 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
 	session->is_stale() ||
 	session->is_killing() ||
 	terminating_sessions) {
-      dout(10) << "currently open|opening|stale|killing, dropping this req" << dendl;
+      dout(10) << "currently open|opening|stale|killing" << dendl;
+      if (m->supported_features.test(CEPHFS_FEATURE_REFRESH_SESSION)) {
+        auto reply = make_message<MClientSession>(CEPH_SESSION_ALREADY_OPEN,
```

Comment:
Did you consider sending `CEPH_SESSION_OPEN` back to the client if the session is already open?

(In client, the case for `CEPH_SESSION_ALREADY_OPEN` falls-through to `CEPH_SESSION_OPEN` anyway)

Reply:
Yeah, let me try this. Mainly different is updating the `seq`.